controller/state/statedb.go

```diff
@@ -483,6 +501,109 @@ func (s *stateDB) countTasks(ctx context.Context) (int, error) {
 	return count, nil
 }
 
+// GetHead gets the latest record update from the controller.
+func (s *stateDB) GetHead(ctx context.Context) (tasks.RecordUpdate, error) {
+	tx, err := s.db().Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	var c string
+	if err := tx.QueryRowContext(ctx, queryHeadSQL, LATEST_UPDATE, "").Scan(&c); err != nil {
+		return nil, err
+	}
+	cidLink, err := cid.Decode(c)
+	if err != nil {
+		return nil, err
+	}
+
+	loader := txContextLoader(ctx, tx)
+	na := tasks.Type.RecordUpdate.NewBuilder()
+	if err = (linksystem.Link{Cid: cidLink}).Load(ctx, ipld.LinkContext{}, na, loader); err != nil {
+		return nil, err
+	}
+
+	if err := tx.Commit(); err != nil {
+		return nil, err
+	}
+
+	return na.Build().(tasks.RecordUpdate), nil
+}
+
+// find all unattached records from worker, collect them into a new record update, and make it the new head.
+func (s *stateDB) PublishRecordsFrom(ctx context.Context, worker string) error {
+	tx, err := s.db().Begin()
+	if err != nil {
+		return err
+	}
+
+	var head string
+	var headCid cid.Cid
+	err = tx.QueryRowContext(ctx, queryHeadSQL, LATEST_UPDATE, "").Scan(&head)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			head = ""
+			headCid = cid.Undef
+		} else {
+			return err
+		}
+	} else {
+		headCid, err = cid.Decode(head)
+		if err != nil {
+			return err
+		}
+	}
+	headCidSig, err := s.PrivKey.Sign([]byte(head))
+	if err != nil {
+		return err
+	}
+
+	itms, err := tx.QueryContext(ctx, queryHeadSQL, UNATTACHED_RECORD, worker)
+	if err != nil {
+		return err
+	}
+
+	rcrds := []tasks.AuthenticatedRecord{}
+
+	for itms.Next() {
+		var lnk string
+		if err := itms.Scan(&lnk); err != nil {
+			return err
+		}
+		if _, err = tx.ExecContext(ctx, updateHeadSQL, ATTACHED_RECORD, lnk); err != nil {
+			return err
+		}
+		sig, err := s.PrivKey.Sign([]byte(lnk))
+		if err != nil {
+			return err
+		}
+		c, err := cid.Decode(lnk)
+		if err != nil {
+			return err
+		}
+		rcrd := tasks.Type.AuthenticatedRecord.Of(c, sig)
+		rcrds = append(rcrds, rcrd)
+	}
+	rcrdlst := tasks.Type.List_AuthenticatedRecord.Of(rcrds)
```

Comment:
reading these names is making my head hurt a little... how about just `records` and `recordList`?

Reply:
in the reputation system parlance we call `Task` a `Record`, and then have a surrounding `AuthenticatedRecord` that is a signature over the cid of the record.